Mathlib/RingTheory/UniqueFactorizationDomain.lean

```diff
@@ -1600,6 +1607,7 @@ noncomputable instance : Lattice (Associates α) :=
     inf_le_left := fun a _ => le_trans (prod_mono inf_le_left) (le_of_eq (factors_prod a))
     inf_le_right := fun _ b => le_trans (prod_mono inf_le_right) (le_of_eq (factors_prod b)) }
 
+open Classical in
```

Comment:
Thanks for removing unnecessary `DecidableEq` instances!

`open classical in` seems fine for definitions, but for theorems we should use the `classical` tactic inside the proof, to be sure that we're not accidentally poisoning the theorem statement with classical instances.